java/sage/IOUtils.java

```diff
@@ -18,12 +18,65 @@
 import sage.io.RemoteSageFile;
 import sage.io.SageInputStream;
 
+import java.io.BufferedReader;
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+
 public class IOUtils
 {
   private IOUtils()
   {
   }
 
+  /**
+   * Used to read a URL contents into String.  This operation blocks and the content is a string, so it should
+   * not be used on large files, and should not be used on binary files.
+   *
+   * @param urlStr URL to download
+   * @return String contents of the given URL
+     */
+  public static String getUrlAsString(String urlStr)
+  {
+    StringBuilder sb = new StringBuilder();
+    try
+    {
+      URL url = new URL(urlStr);
+      InputStream is = url.openStream();
+      BufferedReader br = new BufferedReader(new InputStreamReader(is));
+
+      String line;
+      while ( (line = br.readLine()) != null)
```

Comment:
This will end up stripping off newline characters, so it won't return exactly what is at the other end. You may just want to refactor the getFileAsString method so then you have another one 'getReaderAsString(java.io.Reader) (and if you do that, just get rid of the BufferedReader wrapper and use a char[8192] for the reads...since my current logic in there is sort of silly since it buffers it 8k and then uses 4k reads...so you're doubling all your memcopies)